pkg/controlplane/client.go

```diff
@@ -78,16 +89,86 @@ func (c *client) Authorize(req *api.AuthorizationRequest) (*remoteServerAuthoriz
 	return resp, nil
 }
 
+// IsActive returns if the peer is active or not.
+func (c *client) IsActive() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	return c.active
+}
+
+// setActive the peer status (active or not).
+func (c *client) setActive(active bool) {
+	c.lock.Lock()
+	defer c.lock.Lock()
+	c.active = active
+	if active {
+		c.lastSeen = time.Now()
+	}
+}
+
+// GetHeartbeat get a heartbeat from other peers.
+func (c *client) getHeartbeat() error {
+	var retErr error
+	// copy peer clients array aside
+	peerClients := make([]*jsonapi.Client, len(c.clients))
+	c.lock.RLock()
+	copy(peerClients, c.clients)
+	c.lock.RUnlock()
+
+	for _, client := range peerClients {
+		serverResp, err := client.Get(api.HeartbeatPath)
+		if err != nil {
+			retErr = err
```

Comment:
In the case of wanting to test multiple gateways of a peer, [`errors.Join()`](https://pkg.go.dev/errors@master#Join) may be useful way to collect them.
This applies to adding above `err` as well as `fmt.Errorf()` below